p2p/host/basic/basic_host.go

```diff
@@ -143,17 +150,30 @@ func NewHost(net inet.Network, opts *HostOpts) *BasicHost {
 		h.cmgr = opts.ConnManager
 	}
 
+	var relayCtx context.Context
+	var relayCancel func()
+	if opts.Relay {
+		relayCtx, relayCancel = context.WithCancel(context.Background())
```

Comment:
Added base context as parameter to `NewHost` in https://github.com/libp2p/go-libp2p/pull/210/commits/6be81d34c1016ec8b2ac9143893a1908ae997c08